src/kw_manager.sh

```diff
@@ -265,15 +282,245 @@ function rename_kw_manager_group()
   fi
 }
 
+# This function add a new contact to a given kw mail group
+#
+# @contacts_string: The string with all the contacts informations
+# @group_name: The name of the group which the contacts will be added.
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function add_kw_manager_contacts()
+{
+  local contacts_list="$1"
+  local group_name="$2"
+  local group_id
+  declare -A _contacts_array
+  declare -A _new_contacts_array
+
+  if [[ -z "$contacts_list" ]]; then
+    complain 'The contacts list is empty'
+    return 22 #EINVAL
+  fi
+
+  check_existent_group "$group_name" && exit_msg 'Error while checking group existence'
+
+  group_id="$?"
+
+  if [[ $(split_contact_infos "$contacts_list" '_contacts_array') ]]; then
+    complain 'Error while parsing contacts list'
+  fi
+
+  if [[ $(add_contacts '_contacts_array') ]]; then
+    complain 'Error while adding new contacts to the database'
+  fi
+
+  if [[ $(add_contact_group '_contacts_array' "$group_id") ]]; then
+    complain 'Error while associating contact to the group'
+  fi
+
+  return 0
+}
+
+# This function split the columns in the given contact string passed
+# as parameter and create an associative array with the infos.
+#
+# @contacts_list: An string formed as "CONTACT_NAME <CONTACT_EMAIL>, CONTACT_NAME <CONTACT_EMAIL>, ..."
+# @contacts_array: An associative array formed as "<CONTACT_EMAIL>:<CONTACT_NAME>"
+#
+# Returns:
+# returns 0 if successful, non-zero otherwise
+function split_contact_infos()
+{
+  local contacts_list="$1"
+  local -n contacts_array="$2"
+  local contacts_infos_array
+  local contact_info
+  local ret
+  local name
+  local email
+  local added_contacts=0
+
+  readarray -t contacts_infos_array < <(awk -v RS=", " '{print}' <<< "$contacts_list")
+
+  for contact_info in "${contacts_infos_array[@]}"; do
+    if [[ -z "$contact_info" ]]; then
+      continue
+    fi
+
+    check_infos_sintaxe "$contact_info" || return 22 #EINVAL
+
+    email="$(cut --delimiter='<' --fields=2 <<< "$contact_info" | cut --delimiter='>' --fields=1)"
+    name="$(cut --delimiter='<' --fields=1 <<< "$contact_info")"
+
+    validate_contact_infos "$email" "$name" || return 22 #EINVAL
+
+    contacts_array["$email"]="$name"
+    ((added_contacts++))
+  done
+
+  if [[ "$added_contacts" -ne "${#contacts_array[@]}" ]]; then
+    complain 'Some of the contacts must have a repeated email'
+    return 22 #EINVAL
```

Comment:
Is this a problem if you did not add them? I think if you did not add a duplication, it should be expected that the users will accidentally add a repeated email.